api/health/worker.go

```diff
@@ -117,26 +117,7 @@ func (w *worker) RegisterCheck(name string, check Checker, tags ...string) error
 
 	// If this is a new application-wide check, then all of the registered tags
 	// now have one additional failing check.
-	if isApplicationCheck {
-		// Note: [w.tags] will include AllTag.
-		for tag := range w.tags {
-			w.metrics.failingChecks.WithLabelValues(tag).Inc()
-		}
-		w.numFailingApplicationChecks++
-		return nil
-	}
-
-	// Mark all the tags as failing
-	for _, tag := range tags {
-		gauge := w.metrics.failingChecks.WithLabelValues(tag)
-		gauge.Inc()
-		// If this is the first time this tag was registered, we also need to
-		// account for the currently failing application-wide checks.
-		if w.tags[tag].Len() == 1 {
-			gauge.Add(float64(w.numFailingApplicationChecks))
-		}
-	}
-	w.metrics.failingChecks.WithLabelValues(AllTag).Inc()
+	w.updateMetrics(tc, false, true)
```

Comment:
I like to put comments so primitive args aren't "naked", i.e.

```suggestion
	w.updateMetrics(tc, false /*healthy*/, true /*register*/)
```

Reply:
Same for other calls to `updateMetrics`